dask_planner/src/sql/optimizer.rs

```diff
@@ -40,7 +40,7 @@ impl DaskSqlOptimizer {
     ) -> Result<LogicalPlan, DataFusionError> {
         let mut resulting_plan: LogicalPlan = plan;
         for optimization in &self.optimizations {
-            match optimization.optimize(&resulting_plan, &OptimizerConfig::new()) {
+            match optimization.optimize(&resulting_plan, &mut OptimizerConfig::new()) {
                 Ok(optimized_plan) => resulting_plan = optimized_plan,
                 Err(e) => {
                     return Err(e);
```

Comment:
We'll need to change the code here to be able to skip rules that fail.

```suggestion
                        println!(
                            "Skipping optimizer rule {} due to unexpected error: {}",
                            optimization.name(),
                            e
                        );
```